File: langs/ruby/problems/in_memory_database/stub.rb
class InMemoryDatabase
  def initialize
  end
  def set(key, field, value)
    raise 'not implemented'
  end
  def get(key, field)
    raise 'not implemented'
  end
  def delete(key, field)
    raise 'not implemented'
  end
  def scan(key)
    raise 'not implemented'
  end
  def scan_by_prefix(key, prefix)
    raise 'not implemented'
  end
  def set_at(key, field, value, timestamp)
    raise 'not implemented'
  end
  def set_at_with_ttl(key, field, value, timestamp, ttl)
    raise 'not implemented'
  end
  def delete_at(key, field, timestamp)
    raise 'not implemented'
  end
  def get_at(key, field, timestamp)
    raise 'not implemented'
  end
  def scan_at(key, timestamp)
    raise 'not implemented'
  end
  def scan_by_prefix_at(key, prefix, timestamp)
    raise 'not implemented'
  end
  def backup(timestamp)
    raise 'not implemented'
  end
  def restore(timestamp, timestamp_to_restore)
    raise 'not implemented'
  end
end

File: langs/angular-ts/problems/in_memory_database/stub.ts
class InMemoryDatabase {
  constructor() {}
  set(key, field, value) { throw new Error('not implemented'); }
  get(key, field) { throw new Error('not implemented'); }
  delete(key, field) { throw new Error('not implemented'); }
  scan(key) { throw new Error('not implemented'); }
  scanByPrefix(key, prefix) { throw new Error('not implemented'); }
  setAt(key, field, value, timestamp) { throw new Error('not implemented'); }
  setAtWithTtl(key, field, value, timestamp, ttl) { throw new Error('not implemented'); }
  deleteAt(key, field, timestamp) { throw new Error('not implemented'); }
  getAt(key, field, timestamp) { throw new Error('not implemented'); }
  scanAt(key, timestamp) { throw new Error('not implemented'); }
  scanByPrefixAt(key, prefix, timestamp) { throw new Error('not implemented'); }
  backup(timestamp) { throw new Error('not implemented'); }
  restore(timestamp, timestamp_to_restore) { throw new Error('not implemented'); }
}
module.exports = { InMemoryDatabase };
